pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionUtils.java

```diff
@@ -179,97 +178,91 @@ public static Object getConvertedFinalResult(DataTable dataTable, ColumnDataType
   }
 
   /**
-   * Build a filter operator from the given FilterContext.
-   *
-   * It returns the FilterPlanNode to allow reusing plan level components such as predicate
-   * evaluator map
+   * Build pairs of filtered aggregation functions and corresponding transform operator.
    */
-  public static Pair<FilterPlanNode, BaseFilterOperator> buildFilterOperator(IndexSegment indexSegment,
-      QueryContext queryContext, FilterContext filterContext) {
-    FilterPlanNode filterPlanNode = new FilterPlanNode(indexSegment, queryContext, filterContext);
-    return Pair.of(filterPlanNode, filterPlanNode.run());
-  }
+  public static List<Pair<AggregationFunction[], TransformOperator>> buildFilteredAggregateTransformOperators(
```

Comment:
This method was already huge, so it's a not exactly a new "issue" introduced by this PR, but what are your thoughts on breaking this method into a few logical methods?  Ex. the loop to create the hashmap of `Map<FilterContext, Pair<BaseFilterOperator, List<AggregationFunction>>>` (line 200) could be its own method?  Likewise `// Create the transform operators` on line 230-256?

Reply:
This method is very specific for the filtered aggregate, which groups the aggregates with the same filter and build a transform operator for them.
IMO it is more readable to have all the logic in the same method. If we extract the creation of maps in separate methods, each method will have some assumptions such as main filter not empty, comparing combined filter with main filter etc. which makes them very hard to be reused. In the future if we find part of this method can be reused by another method, we can extract it out at that time.

Reply:
Ya agreed, the resulting methods would not likely be reusable.  I do wonder if having multiple separate methods might help with isolating the intent of each block to aid readability, but I think "readability" is highly subjective.  In any event, I don't think this is new with this PR, and could be altered as a separate, intentional follow up on a need basis.